edge-modules/MetricsCollector/MetricsScrapeAndUpload.cs

```diff
@@ -59,9 +57,16 @@ IEnumerable<Metric> GetGuidTaggedMetrics(IEnumerable<Metric> metrics)
         {
             foreach (Metric metric in metrics)
             {
-                Dictionary<string, string> tagsWithGuid = new Dictionary<string, string>(metric.Tags);
-                tagsWithGuid.Add("guid", this.metricsCollectorRuntimeId.ToString());
-                yield return new Metric(metric.TimeGeneratedUtc, metric.Name, metric.Value, tagsWithGuid);
+                Dictionary<string, string> metricTags = new Dictionary<string, string>(metric.Tags);
+
+                metricTags.Add("testId", this.metricsCollectorRuntimeId.ToString());
```

Comment:
I think you should pass testId and testType using testSpecificTags instead.

Reply:
testId is a Guid that is generated at runtime rather than injected from the pipeline, so I don't think it fits well in testSpecificTags.

My logic for testSpecificTags was to include tags that only exist for a given test type. Since the testType field is used for every test I didn't put it in testSpecificTags.

What do you think?